### PR TITLE
test: Allow self-monitor to be deployed in dev mode

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- allow self-monitor to be deployed in dev mode
- update self-monitor image to use custom image
- basic e2e tests for self-monitor
- Remove some unnecessary config from self-mon deployment
- Relaxed metric verification in e2e tests to `ginkgo.ContainElements` rather than `ginkgo.ConsistsOf` 

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/871

Self-monitor Dockerfile: https://github.com/kyma-project/third-party-images/pull/421

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [x] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [x] The feature is e2e-tested

### CODE REVIEW

File: test/testkit/verifiers/metrics.go [hunk 1]
```diff
@@ -93,3 +93,11 @@ func MetricsFromNamespaceShouldNotBeDelivered(proxyClient *apiserverproxy.Client
 		g.Expect(err).NotTo(HaveOccurred())
 	}, periodic.TelemetryConsistentlyTimeout, periodic.TelemetryInterval).Should(Succeed())
 }
+func MetricPipelineTelemetryHealthFlowIsHealthy(ctx context.Context, k8sClient client.Client, pipelineName string) {
```
Comment (chrkl) [inline test/testkit/verifiers/metrics.go:96]: Add new line above

File: main.go [hunk 2]
```diff
@@ -265,12 +267,12 @@ func main() {
 	flag.StringVar(&fluentBitPriorityClassName, "fluent-bit-priority-class-name", "", "Name of the priority class of fluent bit ")
 
 	flag.BoolVar(&enableSelfMonitor, "self-monitor-enabled", false, "Enable self-monitoring of the pipelines")
-	flag.StringVar(&selfMonitorImage, "self-monitor-image", "quay.io/prometheus/prometheus:v2.45.3", "Image for self-monitor")
+	flag.StringVar(&selfMonitorImageVersion, "self-monitor-image", selfMonitorImage, "Image for self-monitor")
```
[the remark this answers is not in the document]
Comment (rakesh-garimella) [reply main.go]: fixed.

**Review by chrkl — APPROVED** [review]

**Pull request merged**